bootstrap/config/config.go

```diff
@@ -130,15 +141,21 @@ func (cp *Processor) Process(serviceKey string, configStem string, serviceConfig
 
 	switch configProviderInfo.UseProvider() {
 	case true:
-		tokenFile := serviceConfig.GetBootstrap().Service.ConfigAccessTokenFile
-		accessToken, err := token.LoadAccessToken(tokenFile)
-		if err != nil {
-			// access token file doesn't exist means the access token is not needed.
-			if os.IsNotExist(err) {
-				lc.Warnf("Configuration Provider access token at %s doesn't exist. Skipping use of access token", tokenFile)
-			} else {
-				return fmt.Errorf("unable to load Configuration Provider client access token at %s: %s", tokenFile, err.Error())
+		var accessToken string
+
+		// secretProvider will be nil if not configured to be used. In that case, no access token required.
+		if secretProvider != nil {
+			accessToken, err = secretProvider.GetAccessToken(configProviderInfo.serviceConfig.Type, serviceKey)
+			if err != nil {
+				return fmt.Errorf(
+					"failed to get Configuration Provider (%s) access token: %s",
+					configProviderInfo.serviceConfig.Type,
+					err.Error())
 			}
+
+			cp.lc.Infof("Using Config Provider access token of length %d", len(accessToken))
```

Comment:
Out of curiosity, now what we have access token, how is it being cached?

Reply:
Not caching it to reuse between Config and Registry client. They each get their own token. 

The clients are configured with the access tokens, so technically the tokens are cached in each of the clients config.

Reply:
ok.  So each edgeX service nomally gets two consul tokens, one for the registry client, one for the config client?   No response necessary unless I got it wrong.

Reply:
Correct